File: 04web_iot/01led.py
from flask import Flask, request, redirect
import RPi.GPIO as GPIO

app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret'

GPIO.setmode(GPIO.BCM)
pin_led = 18

GPIO.setup(pin_led, GPIO.OUT)

@app.route('/')
def main():
    return redirect('/static/led.html')


@app.route('/operate/<cmd>')
def op(cmd):
    val = request.values['val']   
    if cmd == "led":
        val = request.values['val']   
        print '/operate/', cmd, val 
        if val == 'on':
            GPIO.output(pin_led, True)
            print pin_led, 'on'
        elif val == 'off':
            GPIO.output(pin_led, False)
        return 'OK'

if __name__ == '__main__':
    try:
        app.run(host='0.0.0.0')
    finally:
        print 'cleaning up'
        GPIO.cleanup()

File: 04web_iot/01led.js
var express = require('express');
var app = express();
var onoff =require('onoff');
var pin_led = new onoff.Gpio(18, 'out');

app.use(express.static(__dirname + '/static'));
app.get('/', function(req, res){
	res.redirect('/led.html');
});

app.get('/operate/led', function(req, res){
	console.log(req.url, req.query);
	var val = req.param('val');
	if(val == 'on'){
		pin_led.writeSync(1);
		res.send('OK');
	}else if(val ==='off'){
		pin_led.writeSync(0);
		res.send('OK');
	}else{
		res.send('Fail');
	}
});


app.listen(5000, function(){
	console.log('server running on 5000.');
});